osu.Game/Screens/Play/Player.cs

```diff
@@ -203,9 +203,13 @@ private void load(AudioManager audio, OsuConfigManager config, OsuGameBase game)
             ScoreProcessor.ApplyBeatmap(playableBeatmap);
             ScoreProcessor.Mods.BindTo(Mods);
 
+            dependencies.CacheAs(ScoreProcessor);
+
             HealthProcessor = ruleset.CreateHealthProcessor(playableBeatmap.HitObjects[0].StartTime);
             HealthProcessor.ApplyBeatmap(playableBeatmap);
 
+            dependencies.CacheAs(HealthProcessor);
```

Comment:
Same as in https://github.com/ppy/osu/pull/12670#issuecomment-835057135, can be exposed as a read-only interface.

Reply:
Resolving this to be done in a separate PR.